src/Streamly/Internal/Data/Parser/ParserD/Tee.hs

```diff
@@ -370,6 +374,58 @@ teeWithMin ::
     (a -> b -> c) -> Parser m x a -> Parser m x b -> Parser m x c
 teeWithMin = undefined
 
+-- | Tee a 'Parser' with a 'Fold'
+{-# INLINE teeWithFold #-}
+teeWithFold ::
+       Monad m => (a -> b -> c) -> Parser m x a -> Fold m x b -> Parser m x c
+teeWithFold zf (Parser stepL initialL extractL) (Fold stepR initialR extractR) =
+    Parser step initial extract
+
+    where
+
+    {-# INLINE initial #-}
+    initial = do
+        resL <- initialL
+        resR <- initialR
+        case resL of
+            IPartial sl ->
+                case resR of
+                    Fold.Partial sr -> return $ IPartial (sl, sr)
+                    Fold.Done br -> do
+                        bl <- extractL sl
+                        return $ IDone $ zf bl br
+            IDone bl ->
+                case resR of
+                    Fold.Partial sr -> IDone . zf bl <$> extractR sr
+                    Fold.Done br -> return $ IDone $ zf bl br
+            IError err -> return $ IError err
+
+    {-# INLINE_LATE step #-}
+    step (sl, sr) x = do
+        resL <- stepL sl x
+        case resL of
+            Partial 0 sl1 -> do
+                resR <- stepR sr x
+                case resR of
+                    Fold.Partial sr1 -> return $ Partial 0 (sl1, sr1)
+                    Fold.Done br -> do
+                      bl <- extractL sl1
+                      return $ Done 0 $ zf bl br
+            Continue 0 sl1 -> do
+                resR <- stepR sr x
+                return $ case resR of
+                    Fold.Partial sr1 -> Continue 0 (sl1, sr1)
+                    Fold.Done _ ->
+                        Error $  "teeWithFold: Result of the parser "
+                              ++ "is not extractable."
```

Comment:
Why not extract and let the parser raise the error. Parser knows the best whether it can be extracted or not.